crates/bevy_gizmos/src/primitives/dim3.rs

```diff
@@ -610,13 +610,34 @@ pub struct Cone3dBuilder<'a, 'w, 's, T: GizmoConfigGroup> {
     color: Color,
 
     // Number of segments used to approximate the cone geometry
-    segments: usize,
+    base_segments: usize,
+
+    height_segments: usize,
 }
 
 impl<T: GizmoConfigGroup> Cone3dBuilder<'_, '_, '_, T> {
-    /// Set the number of segments used to approximate the cone geometry.
+    /// Set the number of segments used to approximate the cone geometry for its base and height.
     pub fn segments(mut self, segments: usize) -> Self {
-        self.segments = segments;
+        self.base_segments = segments;
+        self.height_segments = segments;
+        self
+    }
+
+    /// Set the number of segments to approximate the height of the cone geometry.
+    ///
+    /// `segments` should be a multiple of the value passed to [`Self::height_segments`]
+    /// for the height to connect properly with the base.
+    pub fn base_segments(mut self, segments: usize) -> Self {
+        self.base_segments = segments;
+        self
+    }
+
+    /// Set the number of segments to approximate the height of the cone geometry.
+    ///
+    /// `segments` should be a divider of the value passed to [`Self::base_segments`]
```

Comment:
Do you mean `divisor`?
```suggestion
    /// `segments` should be a divisor of the value passed to [`Self::base_segments`]
```